src/git/install.sh

```diff
@@ -61,40 +93,66 @@ receive_gpg_keys() {
     fi
 }
 
-apt_get_update()
-{
-    if [ "$(find /var/lib/apt/lists/* | wc -l)" = "0" ]; then
-        echo "Running apt-get update..."
-        apt-get update -y
+pkg_mgr_update() {
+    if type apt >/dev/null 2>&1; then
+        if [ "$(find /var/lib/apt/lists/* | wc -l)" = "0" ]; then
+            echo "Running apt-get update..."
+            apt-get update -y
+        fi
+    elif [ ${INSTALL_CMD} = "dnf" ] || [ ${INSTALL_CMD} = "yum" ]; then
+        if [ "$(find /var/cache/${INSTALL_CMD}/* | wc -l)" = "0" ]; then
+            echo "Running ${INSTALL_CMD} check-update ..."
+            ${INSTALL_CMD} check-update
+        fi
     fi
 }
 
+
 # Checks if packages are installed and installs them if not
 check_packages() {
-    if ! dpkg -s "$@" > /dev/null 2>&1; then
-        apt_get_update
-        apt-get -y install --no-install-recommends "$@"
+    if type apt >/dev/null 2>&1; then
+        if ! dpkg -s "$@" > /dev/null 2>&1; then
+            pkg_mgr_update
+            apt-get -y install --no-install-recommends "$@"
+        fi
+    elif [ ${INSTALL_CMD} = "dnf" ] || [ ${INSTALL_CMD} = "yum" ]; then
+        _num_pkgs=$(echo "$@" | tr ' ' \\012 | wc -l)
+        _num_installed=$(${INSTALL_CMD} -C list installed "$@" | sed '1,/^Installed/d' | wc -l)
+        if [ ${_num_pkgs} != ${_num_installed} ]; then
+            pkg_mgr_update
+            ${INSTALL_CMD} -y install "$@"
+        fi
+    elif [ ${INSTALL_CMD} = "microdnf" ]; then
+        ${INSTALL_CMD} -y install \
+            --refresh \
+            --best \
+            --nodocs \
+            --noplugins \
+            --setopt=install_weak_deps=0 \
+            "$@"
+    else
+        echo "Linux distro ${ID} not supported."
+        exit 1
     fi
 }
 
 export DEBIAN_FRONTEND=noninteractive
 
-# Source /etc/os-release to get OS info
-. /etc/os-release
+# Debian / Ubuntu packages
 
 # If the os provided version is "good enough", just install that.
 if [ ${GIT_VERSION} = "os-provided" ] || [ ${GIT_VERSION} = "system" ]; then
     if type git > /dev/null 2>&1; then
         echo "Detected existing system install: $(git version)"
         # Clean up
-        rm -rf /var/lib/apt/lists/*
+        clean_up
         exit 0
     fi
 
-    echo "Installing git from OS apt repository"
+    echo "Installing git from OS apt/dnf/microdnf/yum repository"
```

Comment:
Should we do `echo "Installing git from ${INSTALL_CMD} repository"`

Reply:
Added commit to echo either "apt" or "yum/dnf" based on `INSTALL_CMD`. Yum and DNF repositories are, as best as I can tell, the same thing. Yum and DNF are just package managers using the same underlying repository setup.